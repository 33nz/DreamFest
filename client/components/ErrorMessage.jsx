import React from 'react'
import { connect } from 'react-redux'

export const ErrorMessage = (props) => {
  return (
    <div role='alert' className='error'>
      {props.errorMessage}
    </div>
  )
}

function mapStateToProps (state) {
  return {
    errorMessage: state.errorMessage
  }
}

export default connect(mapStateToProps)(ErrorMessage)
