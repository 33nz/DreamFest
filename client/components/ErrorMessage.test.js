import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'

import { ErrorMessage } from './ErrorMessage'

test('renders the error message from props', async () => {
  const errorMessage = 'mock error message'
  const { asFragment } = render(<ErrorMessage errorMessage={errorMessage} />)

  const error = await screen.getByRole('alert')
  expect(error).toHaveTextContent('mock error message')
  expect(asFragment()).toMatchSnapshot()
})
