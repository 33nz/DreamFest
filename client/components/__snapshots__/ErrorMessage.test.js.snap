// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the error message from props 1`] = `
<DocumentFragment>
  <div
    class="error"
    role="alert"
  >
    mock error message
  </div>
</DocumentFragment>
`;
